### PR TITLE
Remove deprecate method and attirbute in `LRScheduler`

### PR DESCRIPTION
**Pull request by zeshengzong**
Following [#99270 suggestion](https://github.com/pytorch/pytorch/issues/99270#issuecomment-1511656408), remove deprecate method `LRScheduler.print_lr`

_____
# BC-breaking note

**`LRScheduler.print_lr()` along with the `verbose` kwarg to the LRScheduler constructor has been deprecated since release 2.2. Please use `LRScheduler.get_last_lr()` to access the learning rate instead.**

`print_lr` and `verbose` were confusing, not properly documented and were little used, as described in #99270, so we deprecated them in 2.2. Now, we complete the deprecation by removing them completely. To access and print the learning rate of a LRScheduler:

In 2.6.0
```
optim = ...
lrsched = torch.optim.lr_scheduler.ReduceLROnPlateau(optim, verbose=True)
// lrsched will internally call print_lr
```

In 2.7.0
```
optim = ...
lrsched = torch.optim.lr_scheduler.ReduceLROnPlateau(optim)
print(lrsched.get_last_lr())
```

### CODE REVIEW

**Comment by vadimkantorov** [comment]
I wonder if `self.verbose` field there can also be removed, it's passed quite a long time since its deprecation

**Comment by janeyx99** [comment]
Thanks for following up on this--yes let's remove everything associated with this deprecation, including the verbose param

**Comment by zeshengzong** [comment]
Should remove `epoch` in `step` method as well? 

https://github.com/pytorch/pytorch/blob/bd370c138a9378d807ad16228cc6a066f14a526d/torch/optim/lr_scheduler.py#L208-L255

**Comment by zeshengzong** [comment]
@janeyx99 Hi, please check changes when available, thanks!

**Comment by janeyx99** [comment]
> Should remove `epoch` in `step` method as well?

no this one is harder to remove

**Review by janeyx99 — APPROVED** [review]
thanks!

**Comment by zeshengzong** [comment]
Seems need help skip BC Lint check, thanks! @janeyx99
